src/OrchardCore/OrchardCore.DisplayManagement/Zones/ZoneShapes.cs

```diff
@@ -18,10 +18,15 @@ public class ZoneShapes : IShapeAttributeProvider
         private const string ContentKey = "Content";
 
         [Shape]
-        public async Task<IHtmlContent> Zone(IDisplayHelper DisplayAsync, IEnumerable<object> Shape)
+        public async Task<IHtmlContent> Zone(IDisplayHelper DisplayAsync, IShape Shape)
         {
+            if (Shape.TryGetProperty("ZoneName", out string zoneName))
+            {
+                Shape.Metadata.Alternates.Add("Zone__" + zoneName);
```

Comment:
Did you try this? 
I don't know that adding an alternate here is the right place, this is essentially the view for the zone, so I think it's being added during the rendering, so it would not take effect. Might need to be done from an `IShapeTableProvider` on displaying

Reply:
You're right. I actually played around with multiple implementations, including an `IShapeTableProvider`, and wanted to simplify it but apparently managed to commit a non-working one. Fixed it.